### PR TITLE
MGMT-13028 - Show only enabled operators in the Review and create summary

### PR DESCRIPTION
**Pull request by jgyselov**
https://issues.redhat.com/browse/MGMT-13028

(The changed test ids shouldn't be an issue - the tests weren't using them anyway.)

### CODE REVIEW

File: src/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx [hunk 1]
```diff
@@ -15,43 +15,26 @@ export const ReviewOperatorsTable = ({ cluster }: { cluster: Cluster }) => {
   const { t } = useTranslation();
   const operatorNames = operatorLabels(t);
 
+  const operators = React.useMemo(
+    () => [OPERATOR_NAME_CNV, OPERATOR_NAME_ODF, OPERATOR_NAME_LVM],
```
Comment (celdrake) [inline src/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx:19]: Ideally I'd want to have a type defined in a similar way to `SupportedPlatformType` that we can define in a single place and drives all operators. 

Therefore if we are missing an entry for an operator in such type, we'd get a TS error and we don't need to actively think about all places where we need to change when a new operator is added.

Comment (jgyselov) [reply src/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx]: I tried my best :slightly_smiling_face:

File: src/common/config/constants.ts [hunk 2]
```diff
@@ -285,21 +285,14 @@ export const OPERATOR_NAME_CNV = 'cnv';
 export const OPERATOR_NAME_LSO = 'lso';
 export const OPERATOR_NAME_ODF = 'odf';
 export const OPERATOR_NAME_LVM = 'lvm';
-export const OPERATOR_NAME_CVO = 'cvo';
```
Comment (jgyselov) [inline src/common/config/constants.ts:288]: I removed the ones (with the corresponding translation stuff) that were unused in the codebase.

File: src/common/config/constants.ts [hunk 3]
```diff
@@ -285,21 +285,14 @@ export const OPERATOR_NAME_CNV = 'cnv';
...
-export const OPERATOR_NAME_CONSOLE = 'console';
```
Comment (jgyselov) [inline src/common/config/constants.ts:289]: I removed the ones (with the corresponding translation stuff) that were unused in the codebase.

File: src/common/config/constants.ts [hunk 4]
```diff
@@ -285,21 +285,14 @@ export const OPERATOR_NAME_CNV = 'cnv';
...
-  | typeof OPERATOR_NAME_CVO
-  | typeof OPERATOR_NAME_CONSOLE;
-
-export const operatorLabels = (t: TFunction): { [key in OperatorName]: string } => ({
-  [OPERATOR_NAME_CONSOLE]: t('ai:OpenShift Console'),
-  [OPERATOR_NAME_CVO]: t('ai:OpenShift Cluster Version Operator'),
-  [OPERATOR_NAME_LSO]: t('ai:OpenShift Local Storage'),
+
+const OperatorNames = [OPERATOR_NAME_CNV, OPERATOR_NAME_LSO, OPERATOR_NAME_ODF, OPERATOR_NAME_LVM];
+export const ExposedOperatorNames = [OPERATOR_NAME_CNV, OPERATOR_NAME_ODF, OPERATOR_NAME_LVM];
```
Comment (jgyselov) [inline src/common/config/constants.ts:290]: If at any time we expose LSO to the users (or stop using `OPERATOR_NAME_LSO` in the code), we can remove the distinction between operators and exposed operators.

File: src/common/config/constants.ts (hunk 3 above)
Comment (celdrake) [reply src/common/config/constants.ts:289]: We should check with CIM as I know there are projects that have references to `assisted-ui-lib`

File: src/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx [hunk 5]
```diff
@@ -16,42 +13,18 @@ export const ReviewOperatorsTable = ({ cluster }: { cluster: Cluster }) => {
...
-              ? 'Enabled'
-              : 'Disabled',
-            props: { 'data-testid': 'openshift-data-foundation' },
-          },
-        ],
-      },
-    ];
-  }, [cluster, operatorNames]);
+    return ExposedOperatorNames.filter((operator) =>
+      hasEnabledOperators(cluster.monitoredOperators, operator),
```
Comment (celdrake) [inline src/ocm/components/clusterConfiguration/review/ReviewOperatorsTable.tsx:17]: You have fixed some Eslint warnings by properly typing the data, please update `package.json`

File: src/common/config/constants.ts [hunk 6]
```diff
@@ -285,21 +285,14 @@ export const OPERATOR_NAME_CNV = 'cnv';
...
+
+export type OperatorName = typeof OperatorNames[number];
+export type ExposedOperatorNameType = typeof ExposedOperatorNames[number];
```
Comment (celdrake) [inline src/common/config/constants.ts:293]: either both are `(Exposed)OperatorName` or `(Exposed)OperatorNameType`

**Review by celdrake — CHANGES_REQUESTED** [review]
Looks good! 
We won't need to remember to update the `ReviewOperatorsTable` when a new operator is added, and we have a more robust code and less warnings :)

**Review by celdrake — APPROVED** [review]
All good, only remaining question is whether it's safe to delete the unused operators

**Review by ammont82 — APPROVED** [review]

**Review by celdrake — APPROVED** [review]

**Pull request merged**